src/SAWScript/Interpreter.hs

```diff
@@ -1057,6 +1085,52 @@ primitives = Map.fromList
     , "currently are 'z3' and 'yices'."
     ]
 
+  , prim "enable_solver_cache" "TopLevel ()"
+    (pureVal enable_solver_cache)
+    Experimental
+    [ "Enable solver result caching. Unless set_solver_cache_path is"
+    , " later called, the cache will not persist between sessions."
+    ]
+
+  , prim "set_solver_cache_path" "String -> TopLevel ()"
+    (pureVal set_solver_cache_path)
+    Experimental
+    [ "Enable solver result caching if it is not already enabled, set a"
+    , " path to use for loading and saving cached results, and save to"
+    , " that path any results cached so far from the current session."
+    ]
+  
+  , prim "clean_solver_cache" "TopLevel ()"
+    (pureVal clean_solver_cache)
+    Experimental
+    [ "Remove all entries in the solver result cache which were created"
+    , " using solver backend version(s) which do not match the version(s)"
+    , " in the current environment."
+    ]
+
+  , prim "print_solver_cache" "String -> TopLevel ()"
+    (pureVal (onSolverCache . printSolverCacheByHex))
+    Experimental
+    [ "Print all entries in the solver result cache whose SHA256 hash"
+    , " keys start with the given string. Thus, given an empty string,"
+    , " all entries in the cache will be printed."
+    ]
+
+  , prim "print_solver_cache_stats" "TopLevel ()"
+    (pureVal (onSolverCache printSolverCacheStats))
+    Experimental
+    [ "Print out statistics about how the solver cache was used, namely"
+    , " how many entries are in the cache, how many insertions into the"
+    , " cache have been made so far this session, and how many times"
+    , " cached results have been used so far this session." ]
+
+  , prim "test_solver_cache_stats" "[Int] -> TopLevel ()"
+    (pureVal (onSolverCache . testSolverCacheStats))
+    Experimental
+    [ "Check whether the values of the statistics printed out by"
+    , " print_solver_cache_stats are equal to those given, failing if"
+    , " this does not hold. Used for testing." ]
```

Comment:
I'm rather confused as to how you figure out what arguments to pass to `test_solver_cache_stats`. The description for `print_solver_cache_stats` suggests that this command has three different statistics, but every call site to `test_solver_cache_stats` is a list with _six_ arguments. What explains the discrepancy there?

Also, is there a good reason for this to be a list? Why not six separate `Int` arguments?

Reply:
Good point, I have absolutely no idea why I implemented this as a `[Int]`. Fixed, and also added clear docs about what values are printed out by `print_solver_cache_stats` and tested by `test_solver_cache_stats`.